src/ApolloClient.ts

```
@@ -374,7 +374,11 @@ export class ApolloClient<TCacheShape> implements DataProxy {
     options: DataProxy.Query<TVariables>,
     optimistic: boolean = false,
   ): T | null {
-    return this.cache.readQuery<T, TVariables>(options, optimistic);
+    try {
+      return this.cache.readQuery<T, TVariables>(options, optimistic);
+    } catch (e) {
+      return null;
```

A blind `try`-`catch` block that swallows all exceptions cannot the right way to do this. This repository controls the cache as well as the client, so there's no reason we should need a band-aid like this.

---

I fully agree, unfortunately I am not familiar enough with the apollo code-base and just wanted to do a suggestion and raise some attention to the implementation of the requested feature as our code-base for a new project is currently suffering from it